free-1.md

```diff
@@ -4,40 +4,130 @@
+completedly untouched beyond a few inital thought experiements.
+
+This idea, which I was calling Binary Star, involves a video game where
+the player controls an increasingly large fleet of space ships through a
+variety of missions. The interesting catch is that the only way to
+control a ship is through a DSL coding interface. Although plans included
+different types of ships having coding styles (i.e. human ships being a
+controlled with something similar to C, Python, or JavaScript while various
+alien ships might use derivatives of Lisp or heavily parallel language,
+like Go), for the context of this course's project, only one of these should
+reasonably be targeted.
+
+Thus, this project targets meeting a need for expressiveness in the context of
+a video game. This means this language should be helping those who play the
+game express their controls. Although the game I envision is closest related to
+a genre of games called real time strategy (RTS), my thoughts are that by
+exposing a programming interface to players, a gamer can theoretically control
+more ships through careful coding than otherwise through the conventional 
+controls of a mouse and keyboard.
+
+Currently, although professional RTS gamers have impressive control over their
+units using only a mouse and keyboard, as seen in
+[this Starcraft video demonstration](https://www.youtube.com/watch?v=YbpCLqryN-Q),
+this type of game play is only achievable after countless hours of practice.
+Furthermore, the game is balanced around a theoretical limit of human control.
+For instance, in Starcraft II it is common knowledge that 66 Terran Siege Tanks
+will defeat 800 Zerglings if charged blindly into the enemy, as shown in
+[this video](https://www.youtube.com/watch?v=SS6totiGZik) -- please ignore the
+voices and low frame rate. That being said, modders, using bots superhuman CPU
+reaction times, have shown that it is possible for 100 Zerglings to defeat
+20 Terran Siege Tanks if individually controlled with enough persision, as
+seen in [this video](https://www.youtube.com/watch?v=IKVFZ28ybQs). Although
+tools to script commands do exist, they are usually done by programmers after
+the fact for fun and are not core functions of the game. In fact, most games
+would consider any kind of computer aided control tool to be a form of cheating
+since it provides users with an unfair advantage. This advantage is made very
+clear in
+[this comparison video](https://www.youtube.com/watch?v=DXUOWXidcY0)
+between a top tier professional player and a bot designed to do the same task.
+
+Furthermore, although compititions exist to design AIs to play the games,
+one such is UC Berkeley's "Berkeley Overmind" (as seen in 
+[this video](https://www.youtube.com/watch?v=fWtpTu9gB20)) which plays
+Starcraft with a superhuman average actions per minute (APM) of 3397 and a
+peak performance over 7000 APM, no game to my knowledge incorperates coding
+natively for fine grained unit control.
```

Comment:
This seems cool and interesting. Is it educational? i.e., is the player supposed to learn how to program through playing the game, or will the game rely on existing knowledge?